tests/glsl-intrinsic/intrinsic-basic.slang

```diff
@@ -0,0 +1,474 @@
+        && equals(mat4x2(0), outerProduct(vec2(0), vec4(0)))
+        && equals(mat3x4(0), outerProduct(vec4(0), vec3(0)))
+        && equals(mat4x3(0), outerProduct(vec3(0), vec4(0)))
+        && equals(mat2(0), transpose(mat2(0)))
+        && equals(mat3(0), transpose(mat3(0)))
+        && equals(mat4(0), transpose(mat4(0)))
+        && equals(mat2x3(0), transpose(mat3x2(0)))
+        && equals(mat3x2(0), transpose(mat2x3(0)))
+        && equals(mat2x4(0), transpose(mat4x2(0)))
+        && equals(mat4x2(0), transpose(mat2x4(0)))
+        && equals(mat3x4(0), transpose(mat4x3(0)))
+        && equals(mat4x3(0), transpose(mat3x4(0)))
+        && float(0) == determinant(mat2(0))
+        && float(0) == determinant(mat3(0))
+        && float(0) == determinant(mat4(0))
+        //&& equals(mat2(0), inverse(mat2(0)))
+        //&& equals(mat3(0), inverse(mat3(0)))
+        //&& equals(mat4(0), inverse(mat4(0)))
+
+        // 8.8. Integer Functions
+        && genUType(0) == uaddCarry(genUType(0), genUType(0), outGenUType) && genUType(0) == outGenUType
+        && genUType(0) == usubBorrow(genUType(0), genUType(0), outGenUType) && genUType(0) == outGenUType
+        && genIType(0) == bitfieldExtract(genIType(0), int(0), int(0))
+        && genUType(0) == bitfieldExtract(genUType(0), int(0), int(0))
+        && genIType(0) == bitfieldInsert(genIType(0), genIType(0), int(0), int(0))
+        && genUType(0) == bitfieldInsert(genUType(0), genUType(0), int(0), int(0))
+        && genIType(0) == bitfieldReverse(genIType(0))
+        && genUType(0) == bitfieldReverse(genUType(0))
+        && genIType(0) == bitCount(genIType(0))
+        && genIType(0) == bitCount(genUType(0))
+        && genIType(0) == findLSB(genIType(0))
+        && genIType(0) == findLSB(genUType(0))
+        && genIType(0) == findMSB(genIType(0))
+        && genIType(0) == findMSB(genUType(0))
+        ;
+}
+
+__generic<let N : int>
+bool dimNTypeFuncs()
+{
+    typealias genFType = vector<float, N>;
+    typealias genDType = vector<double, N>;
+    typealias genIType = vector<int, N>;
+    typealias genUType = vector<uint, N>;
+    typealias genBType = vector<bool, N>;
+    typealias vec      = vector<float, N>;
+    typealias ivec     = vector<int, N>;
+    typealias uvec     = vector<uint, N>;
+    typealias bvec     = vector<bool, N>;
+
+    genFType outGenFType;
+    genDType outGenDType;
+    genIType outGenIType, outGenIType2;
+    genUType outGenUType, outGenUType2;
+
+    bool voidResults = true;
+
+    // 8.8. Integer Functions
+    umulExtended(genUType(0), genUType(0), outGenUType, outGenUType2);
+    voidResults = voidResults && (genUType(0) == outGenUType && genUType(0) == outGenUType2);
+
+    imulExtended(genIType(0), genIType(0), outGenIType, outGenIType);
+    voidResults = voidResults && (genIType(0) == outGenIType && genIType(0) == outGenIType2);
+
+    return voidResults
+        // 8.1. Angle and Trigonometry Functions
```

Comment:
These tests can be improved by using non zero values.

Reply:
The intention is to make sure the function signatures exist.
We will have to add more detailed functional tests especially for the new functions I added.
I was thinking that our new intern can do that.